CMakeLists.txt

```diff
@@ -60,56 +61,68 @@ target_link_libraries(intro PRIVATE project_warnings --coverage)
-add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src
-        ${CMAKE_CURRENT_BINARY_DIR}/googletest-build
-        EXCLUDE_FROM_ALL)
-
-# The gtest/gtest_main targets carry header search path
-# dependencies automatically when using CMake 2.8.11 or
-# later. Otherwise we have to add them here ourselves.
-if (CMAKE_VERSION VERSION_LESS 2.8.11)
-  include_directories("${gtest_SOURCE_DIR}/include")
-endif()
+  ###########################################################
+  # Google Test framework setup: as described at
+  # https://github.com/google/googletest/blob/master/googletest/README.md
+
+  # Download and unpack googletest at configure time
+  configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
+  execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+          RESULT_VARIABLE result
+          WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
+  if(result)
+    message(FATAL_ERROR "CMake step for googletest failed: ${result}")
+  endif()
+  execute_process(COMMAND ${CMAKE_COMMAND} --build .
+          RESULT_VARIABLE result
+          WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
+  if(result)
+    message(FATAL_ERROR "Build step for googletest failed: ${result}")
+  endif()
 
-# Now simply link against gtest or gtest_main as needed. Eg
-add_executable(gtester google_tests/gtest_main.cpp google_tests/GTestExample.cpp google_tests/GTestExample.h)
-target_link_libraries(gtester
-        PRIVATE project_warnings --coverage
-        PUBLIC gtest gtest_main)
-add_test(NAME GoogleTester COMMAND gtester)
+  # Prevent overriding the parent project's compiler/linker
+  # settings on Windows
+  set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+
+  # Add googletest directly to our build. This defines
+  # the gtest and gtest_main targets.
+  add_subdirectory(${CMAKE_CURRENT_BINARY_DIR}/googletest-src
+          ${CMAKE_CURRENT_BINARY_DIR}/googletest-build
+          EXCLUDE_FROM_ALL)
+
+  # The gtest/gtest_main targets carry header search path
+  # dependencies automatically when using CMake 2.8.11 or
+  # later. Otherwise we have to add them here ourselves.
+  if (CMAKE_VERSION VERSION_LESS 2.8.11)
+    include_directories("${gtest_SOURCE_DIR}/include")
+  endif()
 
-# End Google Test framework setup
-###########################################################
-else()   # CPP_STARTER_USE_GOOGLE_TEST
+  # Now simply link against gtest or gtest_main as needed. Eg
+  add_executable(
+          gtester
+          google_tests/gtest_main.cpp
+          google_tests/GTestExample.cpp)
+  target_link_libraries(gtester
+          PRIVATE project_warnings --coverage
+          PUBLIC gtest gtest_main)
+  add_test(NAME GoogleTester COMMAND gtester)
+
+elseif(CPP_STARTER_USE_CATCH2)
+  find_package(Catch2 REQUIRED)
+  add_executable(
+          catch_tester
+          catch_tests/catch_main.cpp
+          catch_tests/catch_test_example.cpp)
+  target_link_libraries(catch_tester
```

Comment:
Wherever you want the executable to have catch2 include path brought in, you need to link the Catch2::Catch2 library.

Reply:
Like in this example https://github.com/catchorg/Catch2/blob/master/docs/cmake-integration.md#usage-1

Reply:
You are correct that I should have added the target_link_libraries() command. I had it in at one point, but I took it out while I was debugging something and forgot to put it back in. 

However, I'm pretty sure that isn't causing the problem with Appveyor. As I understand it, target_link_libraries() affects the link stage. The Visual Studio compiler is won't even produce object files for my catch tests, so the link stage is never reached.